pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/RecordReaderUtils.java

```diff
@@ -67,167 +59,65 @@ public static InputStream getInputStream(File dataFile)
    */
-  public static Object convertSingleValue(FieldSpec fieldSpec, @Nullable Object value) {
+  public static @Nullable Object convertSingleValue(@Nullable Object value) {
     if (value == null) {
       return null;
     }
-    DataType dataType = fieldSpec.getDataType();
-    if (dataType == FieldSpec.DataType.BYTES) {
-      // Avro ByteBuffer maps to byte[]
-      if (value instanceof ByteBuffer) {
-        ByteBuffer byteBufferValue = (ByteBuffer) value;
 
-        // Use byteBufferValue.remaining() instead of byteBufferValue.capacity() so that it still works when buffer is
-        // over-sized
-        byte[] bytesValue = new byte[byteBufferValue.remaining()];
-        byteBufferValue.get(bytesValue);
-        return bytesValue;
-      } else {
-        Preconditions
-            .checkState(value instanceof byte[], "For BYTES data type, value must be either ByteBuffer or byte[]");
-        return value;
-      }
-    }
-    if (value instanceof Number) {
-      Number numberValue = (Number) value;
-      switch (dataType) {
-        case INT:
-          return numberValue.intValue();
-        case LONG:
-          return numberValue.longValue();
-        case FLOAT:
-          return numberValue.floatValue();
-        case DOUBLE:
-          return numberValue.doubleValue();
-        case STRING:
-          return numberValue.toString();
-        default:
-          throw new IllegalStateException("Illegal data type: " + dataType);
-      }
-    }
-    return convertSingleValue(fieldSpec, value.toString());
-  }
+    if (value instanceof ByteBuffer) {
+      ByteBuffer byteBufferValue = (ByteBuffer) value;
 
-  /**
-   * Converts the string value to a single-valued value based on the given field spec.
-   */
-  public static Object convertSingleValue(FieldSpec fieldSpec, @Nullable String stringValue) {
-    if (stringValue == null) {
-      return null;
-    }
-    DataType dataType = fieldSpec.getDataType();
-    // Treat empty string as null for data types other than STRING
-    if (stringValue.isEmpty() && dataType != DataType.STRING) {
-      return null;
+      // Use byteBufferValue.remaining() instead of byteBufferValue.capacity() so that it still works when buffer is
+      // over-sized
+      byte[] bytesValue = new byte[byteBufferValue.remaining()];
+      byteBufferValue.get(bytesValue);
+      return bytesValue;
     }
-    switch (dataType) {
-      case INT:
-        return Integer.parseInt(stringValue);
-      case LONG:
-        return Long.parseLong(stringValue);
-      case FLOAT:
-        return Float.parseFloat(stringValue);
-      case DOUBLE:
-        return Double.parseDouble(stringValue);
-      case STRING:
-        return stringValue;
-      case BYTES:
-        return BytesUtils.toBytes(stringValue);
-      default:
-        throw new IllegalStateException("Illegal data type: " + dataType);
+    if (value instanceof Number) {
+      return value;
     }
+    return value.toString();
```

Comment:
Where we will do the conversion for dataType=bytes, and value is string.

Reply:
I checked that this will be handled within DataTypeTransformer. There are tests for this in the PinotDataTypeTest